src/mono/mono/eventpipe/ep-provider.c

```diff
@@ -356,7 +373,7 @@ provider_invoke_callback (EventPipeProviderCallbackData *provider_callback_data)
 		// the key and the second is the value.
 		// To convert to this format we need to convert all '=' and ';'
 		// characters to '\0', except when in a quoted string.
-		const uint32_t filter_data_len = ep_rt_utf8_string_len (filter_data);
+		const uint32_t filter_data_len = (uint32_t)strlen (filter_data);
```

Comment:
I wonder if `ep_rt_utf8_string_len` should have some name that makes it less likely that we'll make this mistake again.  We should distinguish between the number of bytes and the number of unicode codepoints.

Reply:
Yes, when I found this error, I decided to drop it from shim, to clarify, and remove need to extract length of utf8 string in characters in library code (mostly passing -1 into conversion routines instead).